### PR TITLE
Upload Causal IV case study in education folder

### PR DESCRIPTION
**Pull request by joonho112**
Hello, I've uploaded a case study document titled "Instrumental Variables Analysis of Randomized Experiments with One-Sided Noncompliance." This study is a collaborative work with Profs. Avi Feller and Sophia Rabe-Hesketh from UC Berkeley. The document is a result of our work on the IES project, which you can review here: [IES Project Link](https://ies.ed.gov/funding/grantsearch/details.asp?ID=3317). I would greatly appreciate it if you could approve this merge request. Thank you.

### CODE REVIEW

**Comment by andrjohns** [comment]
Can you remove the two `.DS_Store` files and update the deprecated syntax in your Stan models?

@davkoh would you be able to review this one? You can see the rendered HTML at [this link](https://htmlpreview.github.io/?https://github.com/joonho112/example-models/blob/master/education/causal_iv_one-sided/case_study_02_IV_one-sided.html) (it takes a second to load)

File: education/causal_iv_one-sided/Model_01_CACE_with_Exclusion_Restriction.stan [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    
+    // Complier (assigned to treatment)
+    if(Z[n] == 1 && W[n] == 1){
+      target += log(pi_c) + bernoulli_lpmf(Y[n] | eta_c1) ;
+    }
+    
+    // Never-taker (assigned to treatment)
+    else if(Z[n] == 1 && W[n] == 0){
+      target +=  log(1 - pi_c) + bernoulli_lpmf(Y[n] | eta_n);
+    }
```
Comment (andrjohns) [inline education/causal_iv_one-sided/Model_01_CACE_with_Exclusion_Restriction.stan:45]: It's a bit more efficient to calculate `log(pi_c)` and `log(1-pi_c)` once and then reuse them:
```stan
model {
  real log_pi_c = log(pi_c);
  real log1m_pi_c = log1m(pi_c);
  ...
}
```

**Review by andrjohns — CHANGES_REQUESTED** [review]
A few suggestions for your Stan code

**Comment by joonho112** [comment]
Hello @andrjohns, Thank you for your insights and guidance. I've updated the Stan code in accordance with your suggestions and removed the .DS_Store files. Please take a moment to review the modifications, and let me know if there are any additional adjustments needed prior to the publication of these files.

**Comment by andrjohns** [comment]
> Hello @andrjohns, Thank you for your insights and guidance. I've updated the Stan code in accordance with your suggestions and removed the .DS_Store files. Please take a moment to review the modifications, and let me know if there are any additional adjustments needed prior to the publication of these files.

The CI (model compilation) is still failing because your Stan models are using the old array syntax which has been removed from Stan. See [this section of the manual](https://mc-stan.org/docs/reference-manual/postfix-brackets-array-syntax.html) for guidance on how to update

**Comment by joonho112** [comment]
I have updated all Stan models with the new array syntax.

File: education/causal_iv_one-sided/stan/cace_with_exclusion.stan [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  // Probabilities for the binomial outcome distributions
+  real<lower=0, upper=1> eta_c0;
+  real<lower=0, upper=1> eta_c1;
+  real<lower=0, upper=1> eta_n;
+}  
+
+transformed parameters {
+  // Superpopulation complier average causal effect (CACE)
+  // in per-1000 units
+  real CACE = (eta_c1 - eta_c0) * 10^3;
```
Comment (andrjohns) [inline education/causal_iv_one-sided/stan/cace_with_exclusion.stan:21]: Move to `generated quantities` block

File: education/causal_iv_one-sided/stan/cace_without_exclusion.stan [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  real<lower=0, upper=1> eta_c0;
+  real<lower=0, upper=1> eta_c1;
+  real<lower=0, upper=1> eta_n0;
+  real<lower=0, upper=1> eta_n1;
+} 
+
+transformed parameters {
+  // Super-population average causal effects 
+  real CACE = (eta_c1 - eta_c0) * 10^3;
+  real NACE = (eta_n1 - eta_n0) * 10^3;
```
Comment (andrjohns) [inline education/causal_iv_one-sided/stan/cace_without_exclusion.stan:22]: Move to `generated quantities` block

**Review by andrjohns — CHANGES_REQUESTED** [review]
Great, thanks for the updates! Just a couple of very minor Stan changes remaining and then will leave to @davkoh for final sign-off

**Review by bob-carpenter — COMMENTED** [review]
The .DS_Store files should not be checked in.

**Comment by joonho112** [comment]
I have made the necessary adjustments to the Stan code, transitioning the codes from the "transformed parameters" section to the "generated quantities" section. Additionally, I have removed all .DS_Store files from the repository.

Could you proceed to merge this Pull Request? I am looking forward to the final sign-off from @davkov.

**Pull request merged**